projects/packages/autoloader/tests/php/bin/test-coverage.php

```diff
@@ -0,0 +1,247 @@
+		$line_offset = $class_line - $src_file_map[ $src_file_path ]['class_line'];
+
+		// Record the file in the transformation map.
+		$transformation_map[ $report_file_path ] = array(
+			'src'         => $src_file_path,
+			'line_offset' => $line_offset,
+		);
+	}
+
+	return $transformation_map;
+}
+
+/**
+ * Processes a v9 CodeCoverage report.
+ *
+ * @param SebastianBergmann\CodeCoverage\CodeCoverage $report The report to process.
+ * @return SebastianBergmann\CodeCoverage\CodeCoverage The processed report.
+ */
+function process_coverage_9( $report ) {
+	$data = $report->getData( true );
+
+	// We're going to merge the line coverage from compiled files into the src files.
+	$line_coverage   = $data->lineCoverage();
+	$transformations = get_path_transformation_map( array_keys( $line_coverage ) );
+
+	$removed_files = array();
+	foreach ( $line_coverage as $file => $lines ) {
+		if ( ! isset( $transformations[ $file ] ) ) {
+			continue;
+		}
+
+		// Prepare the transformations we are going to make.
+		$src_file    = $transformations[ $file ]['src'];
+		$line_offset = $transformations[ $file ]['line_offset'];
+
+		// Create a new line coverage mapped to the src file.
+		$new_coverage = array();
+		foreach ( $lines as $line => $coverage ) {
+			$new_coverage[ $src_file ][ $line - $line_offset ] = $coverage;
+		}
+
+		// Merge the coverage since multiple compiled files may map to a single src file.
+		$merge = new SebastianBergmann\CodeCoverage\ProcessedCodeCoverageData();
+		$merge->setLineCoverage( $new_coverage );
+		$data->merge( $merge );
+
+		// Mark the file for removal from the original coverage.
+		$removed_files[] = $file;
+	}
+
+	// Remove all of the files that we've transformed from the coverage.
+	$line_coverage = $data->lineCoverage();
+	foreach ( $removed_files as $file ) {
+		// Make sure the uncovered file does not show up in the report.
+		$report->filter()->excludeFile( $file );
+		unset( $line_coverage[ $file ] );
+	}
+	$data->setLineCoverage( $line_coverage );
+
+	return $report;
+}
+
+/**
+ * Processes the code coverage report and outputs a clover.xml file.
+ */
+function process_coverage() {
+	echo "Aggregating compiled coverage into unified code coverage report\n";
+
+	// We're going to transform the code coverage object into a Clover XML report.
+	$output_file = get_output_file();
+
+	// Since there is no backwards compatibility guarantee in place for the code coverage
+	// object we need to handle it according to each major version independently.
+	$coverage_version = get_coverage_version();
+	$major_version    = substr( $coverage_version, 0, strpos( $coverage_version, '.' ) );
+
+	// We can finally load the report that we're wanting to process.
+	$report = load_report();
+
+	$function = 'process_coverage_' . $major_version;
+	if ( ! function_exists( $function ) ) {
+		echo "No handler defined for major version $major_version\n";
+		die( -1 );
+	}
```

Comment:
No point in loading the report if we can't process it.
```suggestion
	$function = 'process_coverage_' . $major_version;
	if ( ! function_exists( $function ) ) {
		echo "No handler defined for major version $major_version\n";
		die( -1 );
	}

	// We can finally load the report that we're wanting to process.
	$report = load_report();
```